cpp/include/cudf/utilities/type_dispatcher.hpp

```diff
@@ -118,13 +118,15 @@ CUDF_TYPE_MAPPING(timestamp_ns, type_id::TIMESTAMP_NANOSECONDS);
 template <typename T>
 struct type_to_scalar_type_impl {
   using ScalarType = cudf::scalar;
+  using ScalarDeviceType = cudf::detail::scalar_device_view_base;
```

Comment:
Extended the type_to_scalar_type_impl (and scalar_type_t below) to allow code to get the type of the scalar object AND the type of the scalar_device_view object.  Makes using the scalar_device_view easier.

Reply:
Can't we remove this? The base scalar is meant to be used by API writers but scalar_device_view is not. The one use for `ScalarDeviceType` I see in your comparator can do without the default base case as it will always be initialized with a supported type.

So, yes to `ScalarDeviceType` in template specializations, no to putting it in default template.

Reply:
I have removed it from the default template and everything works.